js/app/models/fcEventModel.js

```diff
@@ -135,17 +136,27 @@ app.factory('FcEvent', function(SimpleEvent) {
 		};
 
 		/**
-		 * lock fc event for editing
+		 * Show undo notification for deletion
+		 * @returns {Promise}
 		 */
-		iface.lock = function() {
-			context.lock = true;
-		};
-
-		/**
-		 * unlock fc event
-		 */
-		iface.unlock = function() {
-			context.lock = false;
+		iface.delete = function() {
+			return new Promise(function(resolve, reject) {
+				context.isDeleted = true;
+
+				const timeout = $timeout(function() {
+					if (context.isDeleted) {
+						resolve();
+					}
+				}, 7500);
```

Comment:
Seems good. So undo actions should be in a queue that gets emptied on timeout or on the `beforeunload` event.